group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -671,7 +671,11 @@ ConsumerGroup getOrMaybeCreateConsumerGroup(
             throw new GroupIdNotFoundException(String.format("Consumer group %s not found.", groupId));
         }
 
-        if (group == null || (createIfNotExists && maybeDeleteEmptyClassicGroup(group, records))) {
+        if (group == null) {
+            return new ConsumerGroup(snapshotRegistry, groupId, metrics);
+        } else if (createIfNotExists && maybeDeleteEmptyClassicGroup(group, records)) {
+            log.info("Got an empty classic group {} when getting consumer group. " +
+                "Removed the old classic group and use an empty consumer group instead.", groupId);
```

Comment:
Should we use the format of the other log? `[GroupId {}] Converted the empty classic group to a consumer group.`?